.changeset/old-zebras-teach.md

```diff
@@ -0,0 +1,15 @@
+---
+'astro': major
+---
+
+Updates `astro sync` behavior
```

Comment:
It seems to me like maybe the *main* idea is that Astro no longer relies on `env.d.ts` for configuring TS, but rather moves functionality to `tsconfig.json`? And, as a side effect, `astro sync` behaviour is changed because that file is no longer the recommended place for configuring that stuff?

Would that be a more helpful framing of what's going on here for people?

Reply:
Yes exactly, Although it's still possible to use `env.d.ts`.

```suggestion
Removes `astro sync` update of `src/env.d.ts`
```
Something like that? Pretty bad I'd say :/

Reply:
Maybe something like?

```suggestion
Changes where TypeScript `include` and `exclude` configuration is set, and updates `astro check` behavior to no longer rely on `src/env.d.ts` 
```

Reply:
```suggestion
Changes the default `tsconfig.json` with better defaults, and makes `src/env.d.ts`  optional
```

wdyt?

Reply:
That works!